prow/cmd/deck/main.go

```diff
@@ -697,16 +700,26 @@ func initSpyglass(cfg config.Getter, o options, mux *http.ServeMux, ja *jobs.Job
 	}
 	c, err := storage.NewClient(ctx, options...)
```

Comment:
Perhaps we should flip around the flags to where we use anonymous authentication unless a flag is specified

Reply:
Let's continue the conversation here: https://github.com/kubernetes/test-infra/pull/17183#discussion_r406898027